### PR TITLE
[Bug-Fix] [Certs] Remove additional newline characters added to the PEM files (Enforcer) and process

### LINKED ISSUE #2239: Handling cert validation failures
### Description:

With a cert file updated with an extra newline following cert validation error is thrown in the enforcer. Although the server continued to start due to this certificate not being valid token validation started failing. Hence we should trim the certs when parsing and in cases of failure the failure should reflect in the health of the pod accordingly. 

> [2021-09-27 13:03:59,902][] ERROR - {org.wso2.choreo.connect.enforcer.util.TLSUtils} - Error while adding certificates to the truststore.
java.security.cert.CertificateException: Could not parse certificate: java.io.IOException: Empty input
	at sun.security.provider.X509Factory.engineGenerateCertificate(Unknown Source) ~[?:?]
	at java.security.cert.CertificateFactory.generateCertificate(Unknown Source) ~[?:?]
	at org.wso2.choreo.connect.enforcer.util.TLSUtils.getCertsFromFile(TLSUtils.java:123) ~[org.wso2.choreo.connect.enforcer-0.9.1-SNAPSHOT.jar:0.9.1-SNAPSHOT]
	at org.wso2.choreo.connect.enforcer.util.TLSUtils.updateTruststoreWithMutlipleCertPem(TLSUtils.java:133) ~[org.wso2.choreo.connect.enforcer-0.9.1-SNAPSHOT.jar:0.9.1-SNAPSHOT]
	at org.wso2.choreo.connect.enforcer.util.TLSUtils.lambda$addCertsToTruststore$1(TLSUtils.java:102) ~[org.wso2.choreo.connect.enforcer-0.9.1-SNAPSHOT.jar:0.9.1-SNAPSHOT]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(Unknown Source) [?:?]
	at java.util.stream.ReferencePipeline$2$1.accept(Unknown Source) [?:?]
	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source) [?:?]
	at java.util.Iterator.forEachRemaining(Unknown Source) [?:?]
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Unknown Source) [?:?]
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source) [?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source) [?:?]
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(Unknown Source) [?:?]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(Unknown Source) [?:?]
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source) [?:?]
	at java.util.stream.ReferencePipeline.forEach(Unknown Source) [?:?]
	at org.wso2.choreo.connect.enforcer.util.TLSUtils.addCertsToTruststore(TLSUtils.java:101) [org.wso2.choreo.connect.enforcer-0.9.1-SNAPSHOT.jar:0.9.1-SNAPSHOT]
	at org.wso2.choreo.connect.enforcer.config.ConfigHolder.loadTrustStore(ConfigHolder.java:310) [org.wso2.choreo.connect.enforcer-0.9.1-SNAPSHOT.jar:0.9.1-SNAPSHOT]
	at org.wso2.choreo.connect.enforcer.config.ConfigHolder.<init>(ConfigHolder.java:100) [org.wso2.choreo.connect.enforcer-0.9.1-SNAPSHOT.jar:0.9.1-SNAPSHOT]
	at org.wso2.choreo.connect.enforcer.config.ConfigHolder.getInstance(ConfigHolder.java:108) [org.wso2.choreo.connect.enforcer-0.9.1-SNAPSHOT.jar:0.9.1-SNAPSHOT]
	at org.wso2.choreo.connect.enforcer.server.AuthServer.main(AuthServer.java:67) [org.wso2.choreo.connect.enforcer-0.9.1-SNAPSHOT.jar:0.9.1-SNAPSHOT]
Caused by: java.io.IOException: Empty input
	... 21 more

### Steps to reproduce:
update a truststore certificate with a certificate file with an additional newline at end. Use an token which is validated agains the said certificate endpoint. 

### Environment details (with versions):
- OS:
- Client:
- Env (Docker/K8s):

---

### PR DESCRIPTION
**Pull request by VirajSalaka**
### Purpose
If the certificate has additional newlines, the content will be preprocessed and parsed as expected.

### Issues

Fixes #2239 

### Automation tests
 - Unit tests added: Yes
 - Integration tests added: No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: enforcer-parent/enforcer/src/test/java/org/wso2/choreo/connect/enforcer/util/TLSUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        Iterator<String> aliases =  trustStore.aliases().asIterator();
+        while (aliases.hasNext()) {
+            String alias = aliases.next();
+            X509Certificate cert = (X509Certificate)trustStore.getCertificate(alias);
+            String serialNum = cert.getSerialNumber().toString();
+            Assert.assertTrue(serialNumList.contains(serialNum));
+            serialNumList.remove(serialNum);
+        }
+    }
+}
```
Comment (renuka-fernando) [inline enforcer-parent/enforcer/src/test/java/org/wso2/choreo/connect/enforcer/util/TLSUtilsTest.java:115]: New line?

File: enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/util/TLSUtils.java [hunk 2]
```diff
@@ -108,24 +110,36 @@ public static void addCertsToTruststore(KeyStore trustStore, String filePath) th
...
-                    log.warn("Provided PEM file contains more than one certificate. Hence proceeding with" +
-                            "the first certificate in the File");
+            throws CertificateException, IOException, EnforcerException {
+        String content = new String(Files.readAllBytes(Paths.get(filepath)));
+
+        if (!content.contains(endCertificateDelimiter)) {
+            throw new EnforcerException("Content Provided within the certificate file:" + filepath + "is invalid.");
+        }
+
+        int endIndex = content.lastIndexOf(endCertificateDelimiter) + 25;
```
Comment (renuka-fernando) [inline enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/util/TLSUtils.java:120]: Shall we use a const for `25` may be getting the length of `endCertificateDelimiter`

**Review by pubudu538 — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[failed] dev-deployment : 20211215.32

**Comment by choreo-cicd** [comment]
[succeeded] dev-deployment : 20211215.43

**Pull request merged**